hack/machine_types/machine_types.go

```diff
@@ -193,6 +200,14 @@ func run() error {
 					machine.ECU = stringToFloat32(attributes["ecu"])
 				}
 
+				// AWS VPC CNI plugin-specific maximum pod calculation based on:
+				// https://github.com/aws/amazon-vpc-cni-k8s/blob/f52ad45/README.md
+				enisPerInstance, enisOK := awsutils.InstanceENIsAvailable[attributes["instanceType"]]
+				ipsPerENI, ipsOK := awsutils.InstanceIPsAvailable[attributes["instanceType"]]
+				if enisOK && ipsOK {
+					machine.MaxPods = enisPerInstance*(int(ipsPerENI)-1) + 2
```

Comment:
It might be better to record the raw values for InstanceENIs and InstanceIPs, because MaxPods might be an implementation detail of the aws-cni provider, and I'm not sure if other implementations would have the exact same calculation (e.g. they might reserve another IP per ENI)

Reply:
Yeah, I like that idea, especially since nothing else in `machine_types.go` is implementation-specific. I'll move the calculation itself to `kubelet.go`, which is the only place it's needed.